MicroPython/src/controller/main_controller.py

```diff
@@ -39,13 +44,32 @@ def __init__(self):
         self.last_test_comment = ""
         self.last_test_end_time = None
 
-        web_app.setup(get_measurement_hook=self.get_measurement,
-                      configure_device_hook=self.device_configuration,
-                      configure_aws_hook=self.configure_data_from_terraform,
-                      configure_sensor_hook=self.configure_sensor,
-                      start_test_data_acquisition=self.start_test_data_acquisition_hook,
-                      start_data_acquisition=self.start_data_acquisition_hook,
-                      get_status_hook=self.get_status)
+        # self.tested_connection_cloud = False
+        self.printed_time = False
+        self.got_sensor_data = False
+        self.published_to_cloud = False
+
+        self.cloud_provider = self.get_cloud_provider()
+
+        if config.cfg.cloud_provider == Providers.AWS:
+            web_app.setup(get_measurement_hook=self.get_measurement,
+                        configure_device_hook=self.cloud_provider.device_configuration,
+                        configure_aws_hook=self.cloud_provider.configure_data_from_terraform,
+                        configure_sensor_hook=self.configure_sensor,
+                        start_test_data_acquisition=self.start_test_data_acquisition_hook,
+                        start_data_acquisition=self.start_data_acquisition_hook,
+                        get_status_hook=self.get_status)
+
+        elif config.cfg.cloud_provider == Providers.KAA:
+            web_app.setup(
+                get_measurement_hook=self.get_measurement,
+                configure_device_hook=self.cloud_provider.device_configuration,
+                configure_sensor_hook=self.configure_sensor,
+                start_test_data_acquisition=self.start_test_data_acquisition_hook,
+                start_data_acquisition=self.start_data_acquisition_hook,
+                get_status_hook=self.get_status
+            )
```

Comment:
Is this if/elif clause needed?

Reply:
AWS one requires aws_hook to be supplied but there is no need for it in KAA cloud. I also think it's a bit cleaner to use if/elif as for each cloud service, the setup can be seen independently.